### PR TITLE
Largest bst

### PR DESCRIPTION
**Pull request by AnushkaS09**
finding the largest BST in a given Binary Tree. Using C++
ISSUE #1355

### CODE REVIEW

**Comment by AnushkaS09** [comment]
@plazzy99 @tarun26091999 @rudrakshi99 Please review

File: Tree/LargestBST.cpp [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+bool isbst(BinaryTreeNode<int> *root,int min,int max){
+    if(root==NULL) return true;
+    if(root->data<min||root->data>max) return false;
+    return isbst(root->left,min,root->data-1)&&isbst(root->right,root->data,max);
+}
+bool isBST(BinaryTreeNode<int> *root){
+
+    return isbst(root,INT_MIN,INT_MAX);
```
Comment (rish-singhal) [inline Tree/LargestBST.cpp:40]: give spaces ","

File: Tree/LargestBST.cpp [hunk 2]
```diff
@@ -0,0 +1,105 @@
+/*Find the BST with the maximum heigth i.e. largest BST of the binary tree, which the user will input.*/
```
Comment (rish-singhal) [inline Tree/LargestBST.cpp:1]: add information about the algorithm

File: Tree/LargestBST.cpp [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+
+}
+
+BinaryTreeNode<int>* takeInput() {
+    int rootData;
+    cout<<"enter the root data:";
+    cin >> rootData;
+    if (rootData == -1) {
+        return NULL;
+    }
```
Comment (rish-singhal) [inline Tree/LargestBST.cpp:57]: add appropriate comments in the code explaining the algorithm

**Comment by rish-singhal** [comment]
Add proper indentation, with spaces around >>,<<,= operators

File: Tree/LargestBST.cpp [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+int largestBSTSubtree(BinaryTreeNode<int> *root) {
+    
+    if(isBST(root)) return height(root);
+    return max(largestBSTSubtree(root->left), largestBSTSubtree(root->right));
+   
+}
+
+BinaryTreeNode<int>* takeInput() { //INPUT FUNCTION
+    int rootData;
+    cout<<"enter the root data:";
```
Comment (rish-singhal) [inline Tree/LargestBST.cpp:53]: update it here.

File: Tree/LargestBST.cpp (hunk 3 above)
Comment (rish-singhal) [reply Tree/LargestBST.cpp:57]: this too

File: Tree/LargestBST.cpp [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+    ~BinaryTreeNode() {     //destructor
+        if (left) delete left;
+        if (right) delete right;
+    }
+};
+
+using namespace std;
+
+
+int height(BinaryTreeNode<int> *root)  //returnning heigth of the tree
```
Comment (rish-singhal) [inline Tree/LargestBST.cpp:28]: this is just the information about what it returns or what the function do, but how it does it is missing.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Tree/readme.md [hunk 6]
```diff
@@ -56,6 +56,8 @@ that, this property should be satisfied at every node in the tree.
 * DFS traversal on Binary Tree ----> [C++](/Code/C++/binary_tree_DFS_traversal.cpp)
 * Diameter of Binary Tree ----> [C++](/Code/C++/diameter_of_binary_tree.cpp)
 * Level Order Traversal in BST ----> [C++]() | [Java]() | [Python](/Code/Python/level_order_traversal_binary_tree.py)
+* Largest BST---->[C++](Tree/LargestBST.cpp)
+* Spiral Traversal of Binary Tree ----> [C++](/Code/C++/spiral_traversal_of_binary_tree.cpp) 
 * Right Side View of Binary Tree ----> [Java](/Code/java/Right_Side_View_Of_BinaryTree.java)
 * Lowest common ancestor in a binary tree ----> [Python](/Code/Python/LCA_in_binary_tree.py)
```
Comment (rish-singhal) [inline Tree/readme.md:62]: sort this list

**Review by rish-singhal — CHANGES_REQUESTED** [review]

File: Tree/readme.md [hunk 7]
```diff
@@ -70,6 +72,3 @@ that, this property should be satisfied at every node in the tree.
 * Maximum Topology Short ----> [C++](/Code/C++/Max_Topology_Short.cpp)
 * Left View of  a Binary Tree----> [C++](/Code/C++/left-view.cpp)
 * Bottom-View of a Binary tree ----> [C++](/Code/C++/bottomView.cpp)
```
Comment (rish-singhal) [inline Tree/readme.md:74]: Still it is not sorted.

**Review by rish-singhal — CHANGES_REQUESTED** [review]

**Pull request merged**